mitosheet/mitosheet/step_performers/one_hot_encoding.py

```diff
@@ -0,0 +1,81 @@
+
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+
+import pandas as pd
+from mitosheet.code_chunks.code_chunk import CodeChunk
+from mitosheet.code_chunks.one_hot_encoding_code_chunk import OneHotEncodingCodeChunk
+
+from mitosheet.state import State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.step_performers.utils import add_columns_to_df, get_param
+from mitosheet.types import ColumnID
+
+class OneHotEncodingStepPerformer(StepPerformer):
+    """
+    Allows you to do a one hot encoding of a column.
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'one_hot_encoding'
+
+    @classmethod
+    def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Optional[Dict[str, Any]]]:
+        sheet_index: int = get_param(params, 'sheet_index')
+        column_id: ColumnID = get_param(params, 'column_id')
+
+        # We make a new state to modify it
+        post_state = prev_state.copy() # TODO: update the deep copies
+
+        column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, column_id)
+        df = post_state.dfs[sheet_index]
+        column_index = df.columns.tolist().index(column_header)
+
+        pandas_start_time = perf_counter()
+        new_columns = pd.get_dummies(df[column_header])
+        pandas_processing_time = perf_counter() - pandas_start_time
+
+        # TODO: make sure there isn't overlap here?
```

Comment:
NOTE: right now, there can be bugs if you try to encode on top of something. Is this something we want to address / I should make a spec for?

Reply:
What do you mean on top of something?

Reply:
Exactly the error you found below! So I'll just add an error for it!